### ISSUE TITLE
Add podman configuration guide

### ISSUE THREAD

**Issue by jezekra1** [i-am-bee/bee-stack#39]
Provide guide how to:
- Completely reinstall podman on different platforms, on macos its very difficult to get rid of the old version:
```sh
podman machine stop
podman machine rm
sudo rm -rf /opt/podman
sudo rm -rf /usr/local/podman
sudo rm -rfv /opt/podman
sudo rm /Library/LaunchDaemons/com.github.containers.podman.helper-dev.plist
sudo rm /etc/paths.d/podman-pkg
sudo rm /var/run/podman-helper-*.socket

# Then remove podman desktop application, then install everything again
```

- Section how to debug compose issues - probably not worth investing time as when aim to switch to helm chart soon?
Podman compose implementation checks for "external provider" of docker compose, if found, it will use it no matter what.
To force a correct version, you need to remove all previous `docker-compose`, `podman-compose` commands (also from other runtimes and PATH) and then re-enable it using the setup instructions:
https://podman-desktop.io/docs/compose/setting-up-compose
It's very hard to get rid of the compose, different on each OS and when you get it working properly, it is never updated.

- Section on rootful/rootless machine:
switching to rootful is not well documented, but should be possible to do without removing the VM

We can't be responsible for users setting up their podman properly, it's a job of podman developers to make it easy. If possible, contibute to podman to improve the situation.

**Comment by planetf1** [comment]
Tried on my podman system, but receive error. The actual version check failure is because podman compose versions (current 1.2.0) differs from docker compose versions (2.26). However given the docs do say to use docker compatibility then that's probably ok...

```
➜  bee-stack git:(main) ./bee-stack.sh setup                                                                                                              <<<
❗ERROR: None of compose command versions meets the required version 2.26.
❗       Found versions: podman compose (1.2.0)
❗       Please (re)install a supported runtime
```

Also I suggest the warning text color is changed from yellow. Whilst fine with a dark theme, with light it can be difficult to read ie:
<img width="1174" alt="Screenshot 2024-11-08 at 15 30 20" src="https://github.com/user-attachments/assets/9d12277b-1f1b-423f-a98d-6afc2d57d7eb">

With the docker compose provider installed via the podman UI, the check passes

**Comment by thomassuedbroecker** [comment]
@planetf1 @jezekra1 This worked for me [FYI ](https://suedbroecker.net/2024/11/08/cheetsheet-essential-steps-to-configure-podman-machines/) on MacOS

**Comment by psschwei** [comment]
> probably not worth investing time as when aim to switch to helm chart soon?

+1 I think it's best to focus efforts on getting a helm chart rather than debugging compose issues

**Comment by thomassuedbroecker** [comment]
So it will only run on Kubernetes and what are the requirements for the local cluster?

**Comment by psschwei** [comment]
I think the requirements are still TBD, but in general I'd imagine some minimum version of Kubernetes and some resource requirements around CPUs / memory.

As far as tools to run local Kubernetes, there's a number of them including kind, minikube, colima,  rancher desktop, podman desktop... , and the choice really comes down to personal preference.

**Comment by jezekra1** [comment]
>(current 1.2.0) differs from docker compose versions (2.26). However given the docs do say to use docker compatibility then that's probably ok...

@planetf1 this is not true actually, podman uses whatever external compose provider you have installed. In your case, you need to uninstall `podman-compose` that you probably installed through brew and install the "extension" in the podman desktop UI settings. It is super confusing and annoying https://podman-desktop.io/docs/compose.

**Comment by jeffmaury** [comment]
The recommended way to run compose on Podman is to install docker-compose CLI have it on the PATH and podman compose will find it and run it against the Podman machine with proper configuration

**Comment by jezekra1** [comment]
There is a troubleshooting guide now covering most cases mentioned here
